src/components/calculator/LegalOfficerCaseCost.tsx

```diff
@@ -0,0 +1,450 @@
+        for(const item of this.parameters.items) {
+            const submittable = api.polkadot.tx.logionLoc.addCollectionItem(
+                api.adapters.toLocId(new UUID()),
+                api.adapters.toH256(Hash.of("")),
+                api.adapters.toH256(Hash.of("")),
+                [
+                    api.adapters.toCollectionItemFile({
+                        contentType: Hash.of("text/plain"),
+                        hash: Hash.of(""),
+                        name: Hash.of(""),
+                        size: item.totalFileSize,
+                    })
+                ],
+                null,
+                true,
+                [],
+            );
+            const itemFees = LegalOfficerCaseCost.multiplyFees(
+                await api.fees.estimateAddCollectionItem({
+                    origin,
+                    collectionItemFee: this._collectionFees.collectionItemFee,
+                    numOfEntries: 1n,
+                    submittable,
+                    tokenIssuance: item.tokenSupply,
+                    totSize: item.totalFileSize,
+                }),
+                item.items
+            );
+            total = LegalOfficerCaseCost.addFees(total, itemFees);
+        }
+        return total;
+    }
+
+    private async recordsFees(api: LogionNodeApiClass, origin: string): Promise<Fees> {
+        let total = new Fees({ inclusionFee: 0n });
+        for(const record of this.parameters.records) {
+            const submittable = api.polkadot.tx.logionLoc.addTokensRecord(
+                api.adapters.toLocId(new UUID()),
+                api.adapters.toH256(Hash.of("")),
+                api.adapters.toH256(Hash.of("")),
+                [
+                    api.adapters.toCollectionItemFile({
+                        contentType: Hash.of("text/plain"),
+                        hash: Hash.of(""),
+                        name: Hash.of(""),
+                        size: record.totalFileSize,
+                    })
+                ],
+            );
+            const recordFees = LegalOfficerCaseCost.multiplyFees(
+                await api.fees.estimateAddTokensRecord({
+                    origin,
+                    tokensRecordFee: this._collectionFees.tokensRecordFee,
+                    numOfEntries: 1n,
+                    submittable,
+                    totSize: record.totalFileSize,
+                }),
+                record.records,
+            );
+            total = LegalOfficerCaseCost.addFees(total, recordFees);
+        }
+        return total;
+    }
+
+    static euroToLgnt(amount: number, lgntEuroRate: number): bigint {
+        const converted = amount * lgntEuroRate;
+        return Currency.toCanonicalAmount(Currency.nLgnt(BigInt(Math.ceil(converted))));
+    }
+
+    static lgntToEuro(lgnt: bigint, lgntEuroRate: number): number {
+        return Currency.toPrefixedNumberAmount(lgnt).convertTo(Numbers.NONE).coefficient.toNumber() / lgntEuroRate;
+    }
```

Comment:
Could be useful to define constant `lgntEuro**Cent**Rate`, use integer division and further divide by 100 in order to get amounts rounded to the eurocent.

Reply:
Good point, I'll add that to the SDK.

Reply:
The conversion logic is now part of the SDK, however I did not work on formatting to make numbers more readable as this implies the values to be displayed differently when being edited or viewed, which is not trivial.